site/blog/unlisted/deploy-stable-diffusion.mdx

```diff
@@ -0,0 +1,269 @@
+
+There is a publicly available <a href="https://beta.dreamstudio.ai/dream">tool</a> , it is limited to a maximum of 200 images generation. But because the model is open source, you can download and host your own version of it. There are different approaches to host it, butone easy option is to host in an EC2 GPU instance, using IaSQL to install with one SQL query.
+IaSQL is an [open-source](https://github.com/iasql/iasql) software tool that creates a two-way connection between an unmodified PostgreSQL database and an AWS account so you can manage your infrastructure from a database.
+
+## Pre-requisites
+
+Stable Diffusion will need to run on GPU instances. These are likely not enabled in your AWS account by default, so you may need to increase your quota. This post uses _p_ instances, so you need to check your quota on <a href="https://us-east-2.console.aws.amazon.com/servicequotas/home/services/ec2/quotas">AWS</a> for the region where you want to deploy your model, and check if you have quota for this type. There is a need of 8 vCPUs per instance, so please request 8 vCPUS per each instance you want to deploy. As instances will take some time to terminate while other is started, it is recommended to request quota for at least 2 instances.
+
+<img src="/img/deploy-stable-diffusion/request_quota_increase.png" style={{ maxWidth: 600 }} />
+
+In order to run the model, there is the need to integrate with Nvidia GPU. This does not come out of the box, and needs to have a custom OS image with drivers installed. This is not a trivial process, but there are some AMIs in the marketplace that can help us with that. This post uses an image from marketplace, so a subscription will be needed in order to use it.
+Please visit the <a href="https://aws.amazon.com/marketplace/pp/prodview-64e4rx3h733ru?sr=0-1&ref_=beagle&applicationId=AWS-Marketplace-Console">AWS marketplace</a> and subscribe to the Amazon Linux 2 AMI with NVIDIA TESLA GPU Driver AMI offer.
+
+<img src="/img/deploy-stable-diffusion/subscribe_to_marketplace_ami.png" style={{ maxWidth: 600 }} />
+
+**Please note that you will incur in a cost of $3.06/Hour when using this AMI**
+
+## Stable Diffusion installation
+
+While Stable Diffusion is a complex model with lots of dependencies, there are pre-built Docker images from the community that ship with all the components on place. We will rely on <a href="https://sygil-dev.github.io/sygil-webui/docs/Installation/docker-guide">Sygil WebUI</a> images to deploy our model. These images are ready to be used out of the box, but there is possibility to build and customize the images in case it's needed.
+
+We would need to start by getting the AMI ID for the image on the marketplace, in order to boot images based on that. We can get it with a simple query using IaSQL SDK module:
+
+```sql title="Get AMI ID"
+ -- install modules
+SELECT
+  *
+FROM
+  iasql_install ('aws_sdk');
```

Comment:
I have the feeling this introduces the iasql query too fast. we might need to say something like now, let's connect to our iasql database and execute the following or something like that? at least introductory?

Reply:
yeah, one sentence would be good that links to https://iasql.com/docs